src/main.js

```diff
@@ -284,22 +284,25 @@ function calculateStats(modules) {
                 _Student.gradesCount[parsedModule.grade]++;
                 _Student.totalGrades++;
             }
+
+            let credits = Number(parsedModule.credits);
             if (parsedModule.passed) {
-                let credits = Number(parsedModule.credits);
                 _Student.totalCredits += credits;
                 let moduleType = parsedModule.moduleType;
                 if (moduleType in _Student.creditsByModuleTypeCount) {
                     _Student.creditsByModuleTypeCount[moduleType].current += credits;
                 }
             }
+
             // if cell is empty, Number('') returns 0!
+            // if no credits available don't use in average calculation
             numericMark = Number(parsedModule.mark)
-            if (!isNaN(numericMark) && numericMark != 0) {
-                _Student.totalNumericMarkWithF += numericMark;
-                _Student.numberOfNumericMarksWithF++;
+            if (!isNaN(numericMark) && numericMark != 0 && credits > 0) {
+                _Student.totalNumericMarkWithF += numericMark * credits;
```

Comment:
Fields `totalNumericMarkWithF`, `totalNumericMark`, `numberOfNumericMarksWithF` and `numberOfNumericMarks` have a misleading name now that they represent weighted values instead of absolute values.

Rename to `weightedTotalOfNumericMarks` and `weightedNumberOfNumericMarks` or something like that.

Reply:
I've updated to code to represent those changes